passwd.c

```diff
@@ -93,5 +106,20 @@ int main(void) {
     free(password);
     free(password_confirmation);
 
-    return EXIT_SUCCESS;
+    return ret;
+}
+
+int main(int argc, char **argv) {
+    switch (argc) {
+    case 1:
+        return main_set_password();
+    case 2:
+        if (strcmp(argv[1], "-l") == 0) {
```

Comment:
Please change to use `strncmp(argv[1], "-l", 2)` just to follow good practice

Reply:
I used strcmp since the rest of the code also uses it. Isn't the suggested use of strncmp exactly equivalent in this case? And aren't argv elements guaranteed to be nul-terminated?
But I guess it's good in case something changes.

Reply:
Yeah, no changes for risk of out-of-bounds reads (I think). IMHO a good idea to always use safe functions nevertheless.
Only change would be that any arg `-l*` is then interpreted as `-l` also.

Reply:
I don't think accepting anything starting with `-l` is a good idea though

Reply:
Maybe a comment like this?
```
        if (strcmp(argv[1], "-l") == 0) { // safe since argv[n] and literals are nul-terminated
```